server.go

```diff
@@ -230,9 +231,21 @@ func NewServer(opts ...ServerOption) (*Server, error) {
 		}
 	}
 
+	path, err := expandDirName(s.Holder.Path)
```

Comment:
I thought we could call `expandDirName()` inside `OptServerDataDir()` like this:
```
 func OptServerDataDir(dir string) ServerOption {
 	return func(s *Server) error {
 		path, err := expandDirName(dir)
 		if err != nil {
 			return err
 		}
 		s.Cluster.Path = path
 		s.Holder.Path = path
 		return nil
 	}
 }
```
Does that work?

Reply:
What if `OptServerDataDir` is never called?

Reply:
That's a good point. I forgot that our default data-dir has a `~` in it.

Reply:
It looks like `OptServerDataDir()` is always called when setting up a server, otherwise that value will be an empty string, so I think it's ok to have `expandDirName()` inside `OptServerDataDir()`.

Reply:
We may later have another `Opt*` function which sets the path, then we have to duplicate the expansion code (or just forget to add it). I think it's best to do the expansion after all `Opt*` functions run unless there's a reason not to do so.

Reply:
either we need to add a `dataDir string` field to Server, and do the expansion, `Cluster.Path`, and `Holder.Path` in `NewServer`, or we can handle everything in the `Opt` as travis suggested. I don't think it's too likely that we'll need another option that touches the data dir, but I've been surprised before.

Reply:
Updated